app/src/app/api/daily-reset/route.ts

```diff
@@ -0,0 +1,61 @@
+import { drizzleDB } from "@/server/db";
+import { dailyReset } from "@/drizzle/schema";
+import { eq, gte, and } from "drizzle-orm";
+import { lockWithGameTimer, handleEndpointError } from "@/libs/gamesettings";
+
+export async function GET() {
+  try {
+    // Ensure this function doesn't run too frequently
+    const timerCheck = await lockWithGameTimer(drizzleDB, 3, "h", "daily-reset-check");
```

Comment:
In my commit I replaced this one with a daily timer, which checks if the date has changed.